test/test_prometheus.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+from asab.metrics.prometheus import metric_to_text, get_labels, validate_format
+
+
+class TestPrometheus(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(TestPrometheus, self).__init__(*args, **kwargs)
+
+    def test_metric_to_text(self):
+        input_counter = {
+            "Name": "mycounter",
+            "Tags": {
+                "host": "DESKTOP-6J7LEI1",
+                "unit": "bytes",
+                "help": "The most important counter ever.",
+                "label": "sth",
+                "other_label": "sth_else",
+            },
+            "Values": {"v1": 10, "v2": 5},
+        }
+
+        expected_output = """# TYPE mycounter_v1_bytes counter
+# UNIT mycounter_v1_bytes bytes
+# HELP mycounter_v1_bytes The most important counter ever.
+mycounter_v1_bytes_total{label="sth",other_label="sth_else"} 10
+# TYPE mycounter_v2_bytes counter
+# UNIT mycounter_v2_bytes bytes
+# HELP mycounter_v2_bytes The most important counter ever.
+mycounter_v2_bytes_total{label="sth",other_label="sth_else"} 5"""
+        output = metric_to_text(input_counter, type="counter")
+        self.assertEqual(expected_output, output)
+
+        # missing labels and units
+        input_counter2 = {
+            "Name": "_mycounter",
```

Comment:
Im confused why I cant see this `_mycounter` string in expected output. Is that because of missing labels and units ?

Reply:
I guess the only problem here is that Open Metrics standard bans leading underscores. ("MetricFamily names beginning with underscores are RESERVED and MUST NOT be used unless specified by this standard.") Prometheus requires names without two leading underscores. Thus, I just removed any leading underscores for simplicity.